cmd/controller/app/controllers.go

```diff
@@ -165,19 +162,6 @@ func getAllControllers(mgr manager.Manager, client k8s.Client, informerFactory i
 			}
 			return err
 		},
-		"s2ibinary": func(mgr manager.Manager) error {
-			return mgr.Add(s2ibinary.NewController(client.Kubernetes(),
-				client.KubeSphere(),
-				kubesphereInformer.Devops().V1alpha1().S2iBinaries(),
-				s3Client,
-			))
-		},
-		"s2irun": func(mgr manager.Manager) error {
```

Comment:
You already removed this part. So, what about the code base of `s2irun.NewS2iRunController`. Do we still need to keep them? If you think it should be kept, it would be better if you could let us know the reason.

Reply:
By the way, should we remove the CRDs as well? And please don't forget to test it.

Reply:
Those controllers could be removed, but the CRDs will recreate when we execute the command of `make manifests generate`, the `s2ibinary type` can not remove yet.

Reply:
Yup, currently, the s2i types cannot be removed. But s2i controllers could be removed safely because we would never run them forever in devops-controller.